ethcore/src/client/client.rs

```diff
@@ -556,7 +564,7 @@ impl Client {
 			BlockID::Number(number) => Some(number),
 			BlockID::Hash(ref hash) => self.chain.block_number(hash),
 			BlockID::Earliest => Some(0),
-			BlockID::Latest => Some(self.chain.best_block_number())
+			BlockID::Latest | BlockID::Pending => Some(self.chain.best_block_number()),
```

Comment:
why add the pending variant to `BlockID` at all then? 
RPC requests could handle miner/client split and then client doesn't even have to know that it exists